crates/sui/src/main.rs

```diff
@@ -45,13 +46,27 @@ async fn main() {
 
     let args = Args::parse();
     let _guard = match args.command {
-        SuiCommand::Console { .. }
-        | SuiCommand::Client { .. }
-        | SuiCommand::KeyTool { .. }
-        | SuiCommand::Move { .. } => telemetry_subscribers::TelemetryConfig::new()
-            .with_log_level("error")
-            .with_env()
-            .init(),
+        SuiCommand::Console { .. } | SuiCommand::KeyTool { .. } | SuiCommand::Move { .. } => {
+            telemetry_subscribers::TelemetryConfig::new()
+                .with_log_level("error")
+                .with_env()
+                .init()
+        }
+        SuiCommand::Client {
+            cmd: Some(ReplayTransaction { gas_info, .. }),
+            ..
+        } => {
+            if gas_info {
+                telemetry_subscribers::TelemetryConfig::new()
+                    .with_trace_target("replay")
```

Comment:
suggestion/ clarifying q: something that i wanted to address but never got around to with the target_prefix/ with_target_prefix on telemetry-subscribers is that I suppose the field and method implies that you could set up multiple trace targets, but in reality is a single string. If we want to log on both replay and say ... gas profiler (or something like that), how would we do so? I don't fully understand the whole logging story tbh, so maybe I'm misunderstanding and it's not an issue, but (although outside the bounds of this PR imo) it would be nice to extend telemetry_subscribers to allow someone to set up multiple directives, such as like telemetry_subscribers::TelemetryConfig::new().with_trace_target("replay").with_trace_target("gas_profiler")

Reply:
Yeah, this is possible! I don't see anyone else has ever made use of targets in our codebase, but if we want to emit logs from multiple targets at the same time, we would append another string to the directives separated by a comma. 

If I was to add a new target to the gas profiler, personally on the initialization of the telemetry config that happens in main, once per command, I would check if the command was `ProfileTransaction` and if so initialize with trace_target to be "profile", and also add in trace lines with `target = "profile"` throughout the code.

Reply:
Ah gotcha, thanks for explaining! So that's what directives are for. That's really neat, and thanks for illustrating with the gas profiler example :pray: